### PR TITLE
fix: skip login on demo app

### PR DESCRIPTION
**Pull request by ZeRego**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: #1153

### Scope of the changes (select all that apply):
- [x] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

File: packages/frontend/src/pages/Login.tsx [hunk 1]
```diff
@@ -88,6 +77,17 @@ const Login: FC = () => {
         return <Redirect to={{ pathname: '/' }} />;
     }
 
+    const isDemo = health.data?.mode === LightdashMode.DEMO;
+    if (health.isLoading || isDemo) {
```
Comment (owlas) [inline packages/frontend/src/pages/Login.tsx:81]: Can you explain the conditional logic in these two? I'm very confused 😂

**Review by owlas — CHANGES_REQUESTED** [review]

File: packages/frontend/src/pages/Login.tsx (hunk 1 above)
Comment (ZeRego) [reply packages/frontend/src/pages/Login.tsx:81]: Yeah, is a bit confusing 🤔  
Basically, we want to show a loading page when the health data is loading or when it's the demo app. 
The check inside is so we only auto login once. It checks if it's the demo app and the login request is not loading. If that request is loading it means we already triggered the mutation once.

Comment (ZeRego) [reply packages/frontend/src/pages/Login.tsx:81]: refactoring this into a hook like before

**Review by owlas — APPROVED** [review]
lgtm 🌟

**Pull request merged**